server/services/version.ts

```diff
@@ -96,3 +96,19 @@ export async function createVersion(user: UserDoc, data: CreateVersion) {
 
   return version
 }
+
+export async function updateManagerLastViewed(id: ModelId) {
+  return VersionModel.findOneAndUpdate(
+    { _id: id },
+    { $set: { managerLastViewed: new Date() as DateString } },
```

Comment:
Ah this is where that new type doesn't work. I believe this is similar to `ObjectId`s in that they change when sent via a REST API.

Are we at the point where we'd benefit from having augmented types for the UI, taking into account `Date` and `ObjectId`s changing when being sent via REST.

If we do that then we can correctly type both server and UI.

If this is the way we want to go I'd vote for that being a separate ticket to the work you've done here.

Reply:
I'm happy to leave it as it is for now and explore this as some sort of future tech debt task yes.